src/poliastro/twobody/propagation.py

```diff
@@ -189,6 +192,134 @@ def _kepler(k, r0, v0, tof, *, numiter):
+    n = 0
+    while n < numiter:
+        c_coeff = [
+            8 / 175,
+            8 / 525,
+            1896 / 336875,
+            28744 / 13138125,
+            26248 / 29859375,
+            134584 / 372246875,
+            129802986344 / 857740555546875,
+            55082676858 / 857740555546875,
+            687061097149992 / 24934041427216796875,
+            8033038585237352 / 673219118534853515625,
+            2892031498456202296 / 555405772791254150390625,
+            4093458329892811912 / 1789640823438485595703125,
+            10507274811793509548037032 / 10398126371321703046014404296875,
+            25259289756593760965336 / 56299726292037542523193359375,
+        ]
+
+        C = 0
+        for idx, c_i in enumerate(c_coeff):
+            C += c_i * A_old ** (idx + 2)
+
+        alpha = 1 + 1 / 5 * A_old + C
+        alpha_prime = 1 - 3 / 5 * A_old
+
+        psi = alpha * T_prime
+        eta = alpha_prime * alpha_prime * psi / (alpha - A_old)
+
+        epsilon = 1 + eta * phi0_prime
+        b = np.abs(epsilon + eta * (phi0_2prime + chi0_prime * psi))
+
+        def f(x, epsilon, b):
+            """ This is eq. 5.83 in Battin. """
+            return x ** 3 - 3 * epsilon * x - 2 * b
+
+        # Solving for the roots
+        x = root(f, b, args=(epsilon, b)).x[0]
+
+        if epsilon + eta * (phi0_2prime + chi0_prime * psi) >= 0:
+            xi = 1 + x
+        else:
+            xi = 1 - x
+
+        theta = psi * eta / xi ** 2
+
+        A = gamma0_prime * theta
+
+        if np.abs(A) <= A_m:
+            """ If value A is less than Amax. """
+            if np.abs(A - A_old) <= rtol:
+                # Since 'lambda' is a reserved function, 'lamda' is used
+                lamda = theta / 2 / alpha
+                kappa = alpha_prime * T_prime / xi
+
+                rho = 1 + gamma0_2prime * lamda + phi0 * kappa
+
+                F = 1 - lamda
+                G = (kappa + phi0 * lamda) / beta0
+                Ft = -beta0 * kappa / rho
+                Gt = 1 - lamda / rho
+
+                rf = F * r.to(u.m).value + G * v.to(u.m / u.s).value
+                vf = Ft * r.to(u.m).value + Gt * v.to(u.m / u.s).value
+
+                return rf, vf
+
+            else:
+                A_old = A
+                n += 1
+        else:
+            """
+            A = np.sign(A) * A_m
+            alpha = 1 + 1 / 5 * A_old + C
+            alpha_prime = 1 - 3 / 5 * A_old
+
+            theta_m = A / gamma0_prime
+            lamda_m = theta_m / 2 / alpha
+            B = alpha * alpha_prime / np.sqrt(alpha - A)
+            Tm_prime = B * np.sqrt(theta_m) * (3 + xi0 * theta_m) + 3 * phi0 * lamda_m
+
+            lamda = theta_m / 2 / alpha
+            kappa = alpha_prime * Tm_prime /
+            """
+            n += 1
+
+    return KeyError("SOLUTION NOT FOUND")
+
+
+def gauss(k, r, v, time_of_flight, rtol=1e-8, **kwargs):
+
+    results = [_gauss(k, r, v, tof, rtol, **kwargs) for tof in time_of_flight]
```

Comment:
My recommendation is that you strip the units from `r` and `v` before calling `_gauss`, so you can use lower level functions and, if you're lucky, `njit` all the thing to make it blazing fast :sunglasses: